### PR TITLE
Specify that role tags are not tied to an instance by default

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: website/source/docs/auth/aws-ec2.html.md [hunk 1]
```diff
@@ -1005,8 +1005,13 @@ in its identity document to match the one specified by this parameter.
 <dl class="api">
   <dt>Description</dt>
   <dd>
-    Creates a role tag on the role. Role tags provide an effective way to restrict the
-    capabilities that are set on the role.
+Creates a role tag on the role. Role tags provide an effective way to restrict
```
Comment (jefferai) [inline website/source/docs/auth/aws-ec2.html.md]: Can we remove "effective" here? It's best-effort. I'd much prefer to explicitly say that role tags can can help restrict capabilities to a subset of role capabilities, but that it should not be relied upon as a security primitive as in real AWS environments adding and removing role tags are often a widely distributed privilege. And maybe reminding that an instance should never be tied to a role that, in a worst case scenario, provides more capabilities than the instance should ever be allowed to have.

Comment (vishalnayak) [inline website/source/docs/auth/aws-ec2.html.md]: Yeah, it really is best-effort. Will change..

File: website/source/docs/auth/aws-ec2.html.md [hunk 2]
```diff
@@ -1005,8 +1005,16 @@ in its identity document to match the one specified by this parameter.
 <dl class="api">
   <dt>Description</dt>
   <dd>
-    Creates a role tag on the role. Role tags provide an effective way to restrict the
-    capabilities that are set on the role.
+Creates a role tag on the role, which help in restricting the capabilities that
+are set on the role. Role tags are not tied to any specific ec2 instance unless
+specified explicitly using `instance_id` parameter. By default, role tags are
```
Comment (jefferai) [inline website/source/docs/auth/aws-ec2.html.md]: s/using/using the/

File: website/source/docs/auth/aws-ec2.html.md [hunk 3]
```diff
@@ -1005,8 +1005,16 @@ in its identity document to match the one specified by this parameter.
...
+designed to be used across all instances that satisfies the constraints on the
+role. Regardless of which instances have role tags on them, the restriction
```
Comment (jefferai) [inline website/source/docs/auth/aws-ec2.html.md]: I'd reword this as "Regardless of which instances have role tags on them, capabilities defined in a role tag must be a strict subset of the given role's capabilities."

File: website/source/docs/auth/aws-ec2.html.md [hunk 4]
```diff
@@ -1005,8 +1005,16 @@ in its identity document to match the one specified by this parameter.
...
+that the capabilities of role tags being a subset of capabilities on the role
+is strictly enforced. Note that, since adding and removing a tag is a widely
```
Comment (jefferai) [inline website/source/docs/auth/aws-ec2.html.md]: s/is a/is often a/

File: website/source/docs/auth/aws-ec2.html.md [hunk 5]
```diff
@@ -1005,8 +1005,16 @@ in its identity document to match the one specified by this parameter.
...
+distributed privilege, care needs to be taken to ensure that the instances are
+attached with correct tags to not let them gain more privileges than what were
+intended.
```
Comment (jefferai) [inline website/source/docs/auth/aws-ec2.html.md]: I still think this is a bit wishy-washy. Mostly hard for people to understand how to use correctly. Maybe add one more sentence: "If a role tag is removed the capabilities inherited by the instance will be those defined on the role, so the role should never provide more capabilities than any given instance can be allowed to gain in a worst-case scenario."

**Comment by jefferai** [comment]
👍
